### PR TITLE
Add dhf flow example

### PR DESCRIPTION
**Pull request by paxtonhare**

### CODE REVIEW

File: examples/dhf-flow/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+If you like, you can read up on [how Smart Mastering works](https://marklogic-community.github.io/smart-mastering-core/how-does-it-work/).
+
+Don't get overwhelmed by all the code in this project. To run Smart Mastering from a DHF Harmonize flow all you need to do is this:
+
+```xquery
+import module namespace process = "http://marklogic.com/smart-mastering/process-records"
+  at "/ext/com.marklogic.smart-mastering/process-records.xqy";
+
+process:process-match-and-merge($id)
```
Comment (dmcassel) [inline examples/dhf-flow/README.md]: please add something like "where `$id` is the URI of an entity document that you'd like to find matches for".

File: examples/dhf-flow/gradle-local.properties [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Put your overrides from gradle.properties here
```
Comment (dmcassel) [inline examples/dhf-flow/gradle-local.properties]: this shouldn't be in git. You added it to .gitignore, but I guess not in time

Comment (paxtonhare) [reply examples/dhf-flow/gradle-local.properties]: normally, I agree. But Quickstart requires a gradle-$env in order to show the list of environments. If the file is not there, then we have to add instructions for the end user to add the file so quickstart would work. I felt it better to check it in.  And in the .gitignore it's prefixed with a ! which means it IS allowed :)

cool?

Comment (dmcassel) [reply examples/dhf-flow/gradle-local.properties]: okay, sounds good

**Pull request merged**